src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -88,103 +128,1168 @@ static DamageTable D_80A31B1C = {
+}
+
+/**
+ * Sit idle for one frame, updating rotation and scale to give
+ * the actor an amorphous, oozing effect.
+ * Chuchu will engage player during idle sit if player is visible and
+ * close enough to it.
+ */
+void EnSlime_SitIdle(EnSlime* this, PlayState* play) {
+    f32 sqrtFactor;
+    f32 sinFactor;
+
+    this->timer--;
+    sqrtFactor = sqrtf(this->timer) * 0.2f;
+    EnSlime_UpdateBlink(this);
+
+    sinFactor = ((sin_rad(this->timer * (2.0f * M_PI / 5.0f)) * (0.13f * sqrtFactor)) + 1.0f) * 0.01f;
+    this->actor.scale.x = sinFactor;
+    this->actor.scale.z = sinFactor;
+    if (this->timer < 24) {
+        this->actor.scale.y = ((cos_rad(this->timer * (2.0f * M_PI / 5.0f)) * (0.05f * sqrtFactor)) + 1.0f) * 0.01f;
+    }
+
+    this->actor.shape.rot.x = randPlusMinusPoint5Scaled(0x200) * sqrtFactor;
+    this->actor.shape.rot.z = randPlusMinusPoint5Scaled(0x200) * sqrtFactor;
+
+    // If player is not wearing the stone mask, is within 280 units of the chuchu, and the chuchu
+    //   is facing the player, initialize chase/attack cycle.
+    if ((Player_GetMask(play) != PLAYER_MASK_STONE) && (this->actor.xzDistToPlayer < 280.0f) &&
+        Actor_IsFacingPlayer(&this->actor, 0x5000)) {
+        EnSlime_EngagePlayer(this);
+    } else if (this->timer == 0) {
+        // Done sitting, start moving (remaining idle)
+        EnSlime_ChangeDirectionIdle(this);
+    }
+}
+
+/**
+ * Reset the action timer (to 120), then select a new direction for
+ * chuchu to move.
+ * Set action function to MoveInDirectionIdle
+ */
+void EnSlime_ChangeDirectionIdle(EnSlime* this) {
+    this->timer = 120;
+
+    /*If actor is more than 120 units from its home, set the new direction to be towards its home.
+     * Otherwise, the new direction to move is pretty much random.
+     */
+    if (Actor_XZDistanceToPoint(&this->actor, &this->actor.home.pos) > 120.0f) {
+        this->idleRotY = Actor_YawToPoint(&this->actor, &this->actor.home.pos);
+    } else if (Rand_ZeroOne() < 0.7f) {
+        this->idleRotY = (s32)randPlusMinusPoint5Scaled(0x4000) + this->actor.shape.rot.y;
+    }
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->actionFunc = EnSlime_MoveInDirectionIdle;
+}
+
+/**
+ * While moving idle, update rotation, speed, scale in a sinusoidal fashion to
+ * give oozing effect.
+ * If actor has hit a wall or is too far from home, turn around.
+ * Chuchu will engage player during idle movement if player is visible and
+ * close enough to it.
+ */
+void EnSlime_MoveInDirectionIdle(EnSlime* this, PlayState* play) {
+    f32 sinFactor;
+
+    EnSlime_UpdateBlink(this);
+    this->timer--;
+    Math_ApproachS(&this->actor.shape.rot.y, this->idleRotY, 4, 0x100);
+
+    /* If actor is touching a wall (bgCheckFlag 8?), set target rotation to match wall where it is touching.
+     * If actor is more than 120 units from its home, turn around to face home.
+     */
+    if (this->actor.bgCheckFlags & BGCHECK_CHECK_ONE_FACE) {
```

Comment:
```suggestion
    /* If actor is touching a wall, set target rotation to match wall where it is touching.
     * If actor is more than 120 units from its home, turn around to face home.
     */
    if (this->actor.bgCheckFlags & 8) {
```

The usage of `BGCHECK_CHECK_ONE_FACE` here is incorrect. `BGCHECK_CHECK` flags are meant to be used internally by z_bgcheck.c